File: 0153-find-minimum-in-rotated-sorted-array/solution.java
class Solution {
    public int findMin(int[] nums) {
        int n = nums.length;
        int low = 0;
        int high = n-1;

        while(low < high) {
            int mid = (low+high)/2;
            if(nums[mid] > nums[high]) {
                // go to right
                low = mid+1;
            } else if (nums[mid] <= nums[high]) {
                // go to left
                high = mid;
            }
        }
        return nums[low];
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/solution.swift
class Solution {
    func findMin(_ nums: [Int]) -> Int {
        var n = nums.count
        var low = 0
        var high = n-1
        var ans = 0

        while low < high {
            var mid = low+(high-low)/2
            // right half is sorted
            if nums[mid] < nums[high] {
                high = mid
            } else {
                low = mid+1
            }
        }
        return nums[low]
    }
}
